typed-racket-lib/typed-racket/rep/type-rep.rkt

```diff
@@ -1434,7 +1437,40 @@
          [else default]))
      (type-var-transform initial instantiate-idx)]))
 
-
+(define/cond-contract (abstract-type-in-prop initial names-to-abstract)
+  (-> Prop? (or/c symbol? (listof symbol?)) Prop?)
+  (match initial
+    [(? TypeProp? prop) (TypeProp-update prop type
+                                         (lambda (old-ty)
+                                           (abstract-type old-ty names-to-abstract)))]
+    [tp #:when (or (equal? -tt tp) (equal? -ff tp)) tp]
+  ;; TODO finish the function for other types of props
```

Comment:
What is missing here?

Reply:
Other prop reps like `AndProp` and `OrProp`, but I am planning to do that in future PRs, because dealing with those kinds of props was not a priority in this PR.  

This function and the one above are more of stopgaps, because the way `abstract-*/instantiate-*` is currently organized is okay but could be improved. For example, `instantiate-obj` and `abstract-obj` are put in `type-rep.rkt`. I think it would be more natural to put them in `object-rep.rkt`. Apparently there would be some dependency issues(e.g, `abstract-obj` needs to know some type reps, but `type-rep.rkt` also requires `object-rep.rkt`) , which could be solved by `lazy-require`. But my sense is we could doing instantiation or abstraction via generic interfaces given that the current implementation of Rep structures.